### PR TITLE
feat: quest offering

### LINKED ISSUE #93: [Quests Server] Change `StartQuest` logic for a Quest offer

### PR DESCRIPTION
**Pull request by lauti7**
Closes #93

### CODE REVIEW

File: docs/quests.proto [hunk 1]
```diff
@@ -94,10 +102,18 @@ message QuestStateUpdate {
   QuestState quest_state = 4;
 }
 
+message QuestOffering {
+  string id = 1;
+  string name = 2;
+  string description = 3;
+  QuestDefinition definition = 4;
+}
```
Comment (guidota) [inline docs/quests.proto:110]: wdyt if we reuse the `ProtoQuest` message?
```suggestion
message QuestOffering {
  string id = 1;
  ProtoQuest quest = 2;
}
```

File: crates/server/src/rpc/service.rs [hunk 2]
```diff
@@ -23,6 +25,35 @@ type QuestRpcResult<T> = Result<T, QuestError>;
...
+                            definition: Some(quest.definition),
+                        })),
+                    };
+                    if subscription.r#yield(user_update).await.is_err() {
+                        log::error!("QuestServiceImplementation > StartQuest Error > Not able to send update to susbcription")
+                    }
+                }
+            }
+
+            Ok(QuestOfferResponse { offered: true })
```
Comment (guidota) [inline crates/server/src/rpc/service.rs:51]: I think we should set `offered: true` just for the case when there is a subscription and could send the update. In that case, should we notify the user that they should subscribe first?

**Review by guidota — COMMENTED** [review]
Looking good, just a comment

File: crates/server/src/rpc/service.rs (hunk 2 above)
Comment (lauti7) [reply crates/server/src/rpc/service.rs:51]: I agree but I am not sure how can we handle the notification in order to subscribe

**Pull request closed without merging**